vendor/github.com/newrelic/infra-integrations-sdk/jmx/jmx.go

```diff
@@ -137,22 +187,17 @@ func WithRemoteStandAloneJBoss() Option {
 	}
 }
 
-func openConnection(config *connectionConfig) error {
-	lock.Lock()
```

Comment:
I see the global lock has been removed -- in the Kafka integration, we depend on this when opening connections to multiple brokers (we do it expecting that the JMX package handles synchronization). Does this mean that we now have to handle this from the integration?

Reply:
Specifically, we run `jmx.Open` multiple times concurrently expecting that the lock will make them sequential. It seems this would be broken with these revisions

Reply:
We could restore the lock, but at the moment `jmx` package design only provides global functions not attached to any client/struct, so there's just 1 global var to keep the connection context.
I'm wondering how do you get to handle several connections open at the same time?

Reply:
I was mistaken -- we use our own lock rather than depending on the internal lock to ensure that only one call is executing at a time.